source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -45,8 +45,9 @@ Envoy::Ssl::CertificateDetailsPtr Utility::certificateDetails(X509* cert, const
       std::make_unique<envoy::admin::v3::CertificateDetails>();
   certificate_details->set_path(path);
   certificate_details->set_serial_number(Utility::getSerialNumberFromCertificate(*cert));
-  certificate_details->set_days_until_expiration(
-      Utility::getDaysUntilExpiration(cert, time_source));
+  const auto tmp = Utility::getDaysUntilExpiration(cert, time_source);
+  const auto days_until_expiry = tmp.has_value() ? tmp.value() : -1;
+  certificate_details->set_days_until_expiration(days_until_expiry);
```

Comment:
I'm sorry I didn't realize this before, but I don't think this is correct.

The `days_until_expriation` field in `envoy::admin::v3::CertificateDetails` is a `uint64` (see https://github.com/envoyproxy/envoy/blob/main/api/envoy/admin/v3/certs.proto#L61). I think this is silently converting `-1` to UINT64_MAX.

Reply:
@envoyproxy/api-watchers  Can I change this API? I will do it if get confirmation from you.

Reply:
I want to change the type to `int64`:
https://github.com/envoyproxy/envoy/blob/e33f444b4c5ee783a8a1126f3186832f4a031a39/api/envoy/admin/v3/certs.proto#L61